drop-storage/migrations/20230406110818_initial_migration.sql

```diff
@@ -1,127 +1,142 @@
+-- transfer states
+CREATE TABLE IF NOT EXISTS transfer_active_states (
+  id INTEGER PRIMARY KEY AUTOINCREMENT,
+  transfer_id TEXT NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
   FOREIGN KEY(transfer_id) REFERENCES transfers(id) ON DELETE CASCADE ON UPDATE CASCADE
 );
 
--- traversed paths for outgoing transfer at the time of the transfer creation
--- this is used to "freeze" the state of the files so we could choose to either traverse again or use the traversed paths as origin for the transfer
-CREATE TABLE IF NOT EXISTS outgoing_traversed_paths (
-  id INTEGER PRIMARY KEY AUTOINCREMENT, 
-  path_id INTEGER NOT NULL, 
+CREATE TABLE IF NOT EXISTS transfer_cancel_states (
+  id INTEGER PRIMARY KEY AUTOINCREMENT,
+  transfer_id TEXT NOT NULL,
+  by_peer INTEGER NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+  FOREIGN KEY(transfer_id) REFERENCES transfers(id) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS transfer_failed_states (
+  id INTEGER PRIMARY KEY AUTOINCREMENT,
+  transfer_id TEXT NOT NULL,
+  status_code INTEGER NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+  FOREIGN KEY(transfer_id) REFERENCES transfers(id) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+-- all the paths inside the outgoing transfer
+CREATE TABLE IF NOT EXISTS outgoing_paths (
+  id INTEGER PRIMARY KEY AUTOINCREMENT,
+  transfer_id TEXT NOT NULL,
+  path TEXT NOT NULL,
+  path_id TEXT NOT NULL,
   bytes INT NOT NULL, 
-  full_path TEXT NOT NULL, 
-  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP, 
-  FOREIGN KEY(path_id) REFERENCES outgoing_paths(id) ON DELETE CASCADE ON UPDATE CASCADE
+  created_at TIMESTAMP NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+  FOREIGN KEY(transfer_id) REFERENCES transfers(id) ON DELETE CASCADE ON UPDATE CASCADE
 );
 
 -- all the paths inside the incoming transfer
 CREATE TABLE IF NOT EXISTS incoming_paths (
-  id INTEGER PRIMARY KEY UNIQUE NOT NULL, 
+  id INTEGER PRIMARY KEY AUTOINCREMENT,
   transfer_id TEXT NOT NULL,   
   path TEXT NOT NULL, 
+  path_id TEXT NOT NULL,
   bytes INT NOT NULL, 
-  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP, 
+  created_at TIMESTAMP NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
   FOREIGN KEY(transfer_id) REFERENCES transfers(id) ON DELETE CASCADE ON UPDATE CASCADE
 );
 
 -- states for outgoing paths(uploads)
-CREATE TABLE IF NOT EXISTS outgoing_traversed_path_pending_states (
+CREATE TABLE IF NOT EXISTS outgoing_path_pending_states (
   id INTEGER PRIMARY KEY AUTOINCREMENT, 
-  path_id TEXT NOT NULL,
-  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP, 
-  FOREIGN KEY(path_id) REFERENCES outgoing_traversed_paths(id) ON DELETE CASCADE ON UPDATE CASCADE
+  path_id INTEGER NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
```

Comment:
I suggest using a different name. `path_id` is used in the table above but it's a `TEXT`. Where is an `INTEGER`

Reply:
renamed `path_id` to `path_hash` to better convey the value and to keep the naming in line with `transfer_id`